pyanaconda/modules/payloads/source/rpm_mount/rpm_mount.py

```diff
@@ -0,0 +1,145 @@
+#
+# Kickstart module for RPM mount payload source.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from pyanaconda.core.i18n import _
+from pyanaconda.core.signal import Signal
+from pyanaconda.modules.common.structures.payload import RepoConfigurationData
+from pyanaconda.modules.payloads.constants import SourceType, SourceState
+from pyanaconda.modules.payloads.source.source_base import PayloadSourceBase, \
+    RPMSourceMixin
+from pyanaconda.modules.payloads.source.rpm_mount.rpm_mount_interface import \
+    RPMMountSourceInterface
+from pyanaconda.modules.payloads.source.rpm_mount.initialization import \
+    SetUpRPMMountSourceTask
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+
+class RPMMountSourceModule(PayloadSourceBase, RPMSourceMixin):
+    """The RPM mount point source payload module.
+
+    This source will use existing mount point as the payload source. There will be no unmount and
+    mounting involved.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._path = ""
+        self.path_changed = Signal()
+        self._is_ready = False
+
+    def __repr__(self):
+        return "Source(type='{}', path='{}')".format(
+            self.type.value,
+            self._path)
+
+    def for_publication(self):
+        """Get the interface used to publish this source."""
+        return RPMMountSourceInterface(self)
+
+    @property
+    def type(self):
+        """Get type of this source."""
+        return SourceType.RPM_MOUNT
+
+    @property
+    def description(self):
+        """Get description of this source."""
+        return _("RPM Mount {}".format(self._path))
```

Comment:
The CDROM source uses "Local media", so I would use "Local directory". The current description is too confusing.

Reply:
Agree, another thing I wanted to change - especially when I saw it on the Source spoke status.